cylc/uiserver/data_store_mgr.py

```diff
@@ -144,13 +143,23 @@ def update_workflow_data(self, topic, delta, w_id):
             delta, 'time', getattr(delta, 'last_updated', 0.0))
         # If the workflow has reloaded recreate the data
         # otherwise apply the delta if it's newer than the previously applied.
-        if delta.reloaded:
-            self.data[w_id][topic] = {ele.id: ele for ele in delta.deltas}
-            self.data[w_id]['delta_times'][topic] = delta_time
-        elif delta_time >= self.data[w_id]['delta_times'][topic]:
-            apply_delta(topic, delta, self.data[w_id])
-            self.data[w_id]['delta_times'][topic] = delta_time
-            self.reconcile_update(topic, delta, w_id)
+        try:
+            if delta.reloaded:
+                if hasattr(delta, 'last_updated'):
+                    self.data[w_id][topic].CopyFrom(delta)
```

Comment:
So if `delta.last_updated`, then we just use the protobuf [`CopyFrom`](https://developers.google.com/protocol-buffers/docs/pythontutorial#standard-message-methods).

Do we need to update the comment block above? :point_up:

Reply:
`last_updated` is a attribute of `PbWorkflow`, and the CopyFrom clears the existing data (from the respective item in the store), and merges in the new values..

@hjoliver - "Hold the door" on merging, there may be a more explicit way of finding out what delta it is (don't want to use `topic`, as that is about to change)..

Reply:
Ok good to go now.